src/transformers/models/rt_detr/modeling_resnet_rt_detr.py

```diff
@@ -0,0 +1,901 @@
+"""PyTorch ResNet
+
+This started as a copy of https://github.com/pytorch/vision 'resnet.py' (BSD-3-Clause) with
+additional dropout and dynamic global avg/max pool.
+
+ResNeXt, SE-ResNeXt, SENet, and MXNet Gluon stem/downsample variants, tiered stems added by Ross Wightman
+
+Copyright 2019, Ross Wightman
+"""
+
+import math
+from functools import partial
+from typing import Any, Dict, List, Optional, Tuple, Type, Union
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from transformers.models.timm_backbone.configuration_timm_backbone import TimmBackboneConfig
+
+from ...modeling_outputs import BackboneOutput
+from ...modeling_utils import PreTrainedModel
+from ...utils import is_timm_available, is_torch_available, requires_backends
+from ...utils.backbone_utils import BackboneMixin
+
+
+if is_timm_available():
+    import timm
+    from timm.layers import (
+        AvgPool2dSame,
+        DropBlock2d,
+        DropPath,
+        create_attn,
+        create_classifier,
+        get_act_layer,
+        get_norm_layer,
+    )
+    from timm.models._builder import build_model_with_cfg
+    from timm.models._features import feature_take_indices
+    from timm.models._manipulate import checkpoint_seq
```

Comment:
I checked the current implementation of the backbone, but this wouldn't be allowed. 

If a custom ResNet backbone would need to be implemented for RT-DETR, then it would need to be implemented in a similar way to https://github.com/huggingface/transformers/blob/main/src/transformers/models/resnet/modeling_resnet.py, which means the following classes would need to be defined:
- RTDetrResNetLayer
- RTDetrResnetEncoder
- RTDetrResnetBackbone
- etc.

without relying on the timm dependency anymore. This is because the Transformers library is not a modular toolbox where we mix and match various layers in a single model, the entire ResNet backbone of RT-DETR would need to be defined in a single standalone `modeling_rtdetr_resnet.py` script (which copies as much as possible from `modeling_resnet.py` but with the necessary changes).

Reply:
@NielsRogge I fully understand the explanation. I used this module because we can keep the convert_script as simple as possible

Reply:
Convert scripts should reflect the transformers architecture, rather than the other way around. It's OK if we need to update checkpoints already on the hub, as the model is not yet released

Reply:
@amyeroberts @NielsRogge That is true, I will change this resnet part to niels's suggestion :)